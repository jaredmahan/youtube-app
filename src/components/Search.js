import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const SearchWrapper = styled.div`
  display: block;
  border: 1px solid #ccc;
  padding: 10px;
  background-color: #333;
`;
const InputWrapper = styled.div`
  display: inline-block;
  vertical-align: middle;
  & > select {
    padding: 5px;
    border: none;
    border-left: 1px solid #ccc;
    border-radius: 0 2.5px 2.5px 0;
    padding-right: 10px;
  }
`;
const Input = styled.div`
  border-radius: 2.5px 0 0 2.5px;
  display: inline-block;
  background-color: #fff;
  & > input[type="text"] {
    padding: 2.5px 5px;
    margin: 1px;
    border: none;
    height: 19px;
    width: 300px;
  }
`;

export function Search(props) {
  return (
    <SearchWrapper>
      <InputWrapper>
        <Input>
          <input
            type="text"
            placeholder="Begin typing to search..."
            onChange={props.queryChange}
          />
        </Input>
      </InputWrapper>
      <InputWrapper>
        <select value={props.order} onChange={props.orderChange}>
          <option>relevance</option>
          <option>date</option>
          <option>rating</option>
        </select>
      </InputWrapper>
    </SearchWrapper>
  );
}

Search.propTypes = {
  order: PropTypes.string,
  queryChange: PropTypes.func,
  orderChange: PropTypes.func
};

export default Search;
