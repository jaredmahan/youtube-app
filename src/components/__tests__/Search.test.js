import React from "react";
import { Search } from "../Search";
import ShallowRenderer from "react-test-renderer/shallow";

describe("<Search>", () => {
  it("renders correctly", () => {
    const renderer = new ShallowRenderer();
    const props = {
      query: "query",
      order: "order",
      queryChange: () => null,
      orderChange: () => null
    };
    renderer.render(<Search {...props} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
