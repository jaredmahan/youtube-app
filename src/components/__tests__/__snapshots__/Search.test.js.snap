// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search> renders correctly 1`] = `
<ForwardRef(styled.div)>
  <ForwardRef(styled.div)>
    <ForwardRef(styled.div)>
      <input
        onChange={[Function]}
        placeholder="Begin typing to search..."
        type="text"
      />
    </ForwardRef(styled.div)>
  </ForwardRef(styled.div)>
  <ForwardRef(styled.div)>
    <select
      onChange={[Function]}
      value="order"
    >
      <option>
        relevance
      </option>
      <option>
        date
      </option>
      <option>
        rating
      </option>
    </select>
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;
